common/math/geometry/src/polygon/line_segment.cpp

```diff
@@ -120,7 +120,17 @@ auto LineSegment::getPose(const double s, const bool denormalize_s, const bool f
  */
 auto LineSegment::isIntersect2D(const geometry_msgs::msg::Point & point) const -> bool
 {
-  return getIntersection2DSValue(point, true) ? true : false;
+  const bool x_outside = ((point.x - start_point.x) * (end_point.x - point.x) < 0.0);
+  const bool y_outside = ((point.y - start_point.y) * (end_point.y - point.y) < 0.0);
+  if (x_outside || y_outside) {
+    return false;
+  }
+  const double dx = end_point.x - start_point.x;
```

Comment:
`dx`, `dy`, `squared_length` - i think it is worth adding them as attributes of the `LineSegment` class - they are commonly used and it is possible to create them once in the constructor